packages/fabric/src/index.ts

```diff
@@ -0,0 +1,169 @@
+import { Restroom } from "@restroom-mw/core";
+import { ObjectLiteral } from "@restroom-mw/core/src/types";
+import * as crypto from 'crypto';
+// How to include this?
+// import { combineDataKeys } from "@restroom-mw/sawroom/src/lib";
+import { Gateway, connect, Contract, Identity, Signer, signers,
+         Network }
+   from '@hyperledger/fabric-gateway';
+import { Client, credentials } from '@grpc/grpc-js';
+import axios from "axios";
+import { NextFunction, Request, Response } from "express";
+import {
+  ADDRESS,
+  CONNECT,
+  CHANNEL,
+  CONTRACT,
+  QUERY,
+  SUBMIT
+} from "./actions";
+import { TextDecoder } from 'util';
+
+let fabricAddress: string = null;
+let client: Client = null;
+let tlsCertificate;
+let input: ObjectLiteral = {};
+let gateway: Gateway = null;
+let network: Network = null;
+let contract: Contract = null;
+
+const utf8Decoder = new TextDecoder();
+
+
+// The steps in FabricInterop are all required (in this order...)
+enum FabricInterop {
+  Address = 0,
+  Connect,
+  Channel,
+  Contract
+}
+let current: FabricInterop = FabricInterop.Address;
+
+const validateStep = (requested: FabricInterop) => {
+  if(requested > current) {
+    throw new Error(
+`One step is missing
+1. Set endpoint address
+2. Connect organization
+3. Set channel
+4. Set contract (chaincode)`)
+  }
+}
+
+// Copied from sawroom lib
+export const combineDataKeys = (data: ObjectLiteral, keys: string) => {
+  try {
+    return { ...data, ...JSON.parse(keys) };
+  } catch (e) {
+    throw new Error("Keys or data in wrong format");
+  }
+};
+
+export default async (req: Request, res: Response, next: NextFunction) => {
+  const rr = new Restroom(req, res);
+
+  try {
+    rr.onBefore(async (params) => {
+      const { zencode, keys, data } = params;
+      input = combineDataKeys(data, keys);
+
+      const namedParamsOf = (sid: string): string[] => {
+        if (!zencode.match(sid)) return [];
+        const params = zencode.paramsOf(sid);
+        return params.reduce((acc: string[], p: string) => {
+          acc.push(input[p] || p);
+          return acc;
+        }, []);
+      };
```

Comment:
this seems another small function that is probably useful also in other places and wants to be moved in commonplace to be used, by other packages